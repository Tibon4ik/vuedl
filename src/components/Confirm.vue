<template>
  <div class="confirm-box">
    <div v-if="!!title" class="confirm-title">
      <h1 class="title">{{ title }}</h1>
    </div>
    <div class="confirm-message" v-html="message" />
    <div class="confirm-buttons">
      <div class="column">
        <button v-for="button in parsedButtons" :key="button.key"
          class="button is-large is-is-default is-fullwidth"
          @click="onButtonClick(button)"
          v-html="button.text" />
      </div>
    </div>
  </div>
</template>
<script>

import ConfirmMixin from '../mixins/confirm'
export default {
  layout: 'default',
  mixins: [ConfirmMixin]
}

</script>

<style>
  .confirm-box {
    background: #ffffff;
  }
  .confirm-box .confirm-title {
    padding: 20px
  }
  .confirm-box .confirm-message {
    padding: 0 20px;
  }
  .confirm-box .confirm-buttons {
    text-align:  right;
    padding: 10px 20px;
  }
  .confirm-box .confirm-buttons .button {
    font-size: 20px;
    padding-left: 20px;
  }
</style>
