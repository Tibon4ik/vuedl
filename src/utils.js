/*
 * vuedl
 *
 * (c) Savaryn Yaroslav <yariksav@gmail.com>
 *
 * Some functions was imported from nuxt.js/lib/app/utils.js
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
*/

export function destroyVueElement (vm) {
  if (vm && !vm._isDestroyed && (typeof vm.$destroy === 'function')) {
    vm.$destroy()
  }
}

export function findContainer (container) {
  let found
  if (typeof container === 'string') {
    found = document.querySelector(container)
  } else {
    found = container
  }
  if (!found) {
    found = document.body
  }
  return found
}

// todo
// export function middlewareSeries (promises, appContext) {
//   if (!promises.length || appContext._redirected || appContext._errored) {
//     return Promise.resolve()
//   }
//   return promisify(promises[0], appContext)
//     .then(() => {
//       return middlewareSeries(promises.slice(1), appContext)
//     })
// }
