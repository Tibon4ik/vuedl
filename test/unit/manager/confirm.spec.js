import Dialog from '../../../src/dialog'
import DialogManager from '../../../src/manager'
import Confirm from '../../../src/components/Confirm'
import { wrap } from '../../utils'
import Returnable from '../../../src/mixins/returnable'
import Vue from 'vue'

describe('manager', () => {
  let manager

  test('Create manager instance', () => {
    manager = new DialogManager({ context: { store: {} } })
    expect(manager).toBeInstanceOf(DialogManager)
    manager.component('confirm', Confirm)
  })

  test('returnable', async () => {
    const dlg = new Dialog({
      mixins: [ Returnable ],
      template: '<p></p>'
    })
    await dlg.show()
    expect(typeof dlg.vmd.return).toBe('function')
    expect(dlg.element).toMatchSnapshot()
    setTimeout(() => {
      dlg.vmd.return(123)
    }, 5)
    let res = await dlg.wait()
    expect(res).toBe(123)
    expect(document.body.innerHTML).toBe('')
  })

  test('Check confirm', async () => {
    let dlg = await manager.confirm({
      text: 'test',
      actions: ['ok', 'cancel']
    })
    const wrapper = wrap(dlg.vm)
    expect(dlg.vm.$el).toMatchSnapshot()
    setTimeout(() => {
      wrapper.find('[action-key=ok]').trigger('click')
    }, 5)
    let res = await dlg.wait()
    expect(res).toBe('ok')
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Check confirm with btns true|false', async () => {
    let dlg = await manager.confirm({
      text: 'test',
      actions: { 'true': 'Yes', false: 'No' }
    })
    const wrapper = wrap(dlg.vm)
    expect(dlg.element).toMatchSnapshot()
    expect(wrapper.find('[action-key=true]').exists()).toBe(true)
    expect(wrapper.find('[action-key=false]').exists()).toBe(true)
    setTimeout(() => {
      wrapper.find('[action-key=true]').trigger('click')
    }, 10)
    let res = await dlg.wait()
    expect(res).toBe(true)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Check confirm with handle functions', async () => {
    let dlg = await manager.confirm({
      text: 'test',
      actions: {
        false: 'No',
        true: {
          text: 'Yes',
          handle: () => {
            return new Promise((resolve) => {
              setTimeout(() => resolve({ msg: 'foo' }), 10)
            })
          }
        }
      }
    })
    const wrapper = wrap(dlg.vm)
    expect(dlg.element).toMatchSnapshot()
    setTimeout(() => {
      wrapper.find('[action-key=true]').trigger('click')
      Vue.nextTick(() => {
        expect(wrapper.find('[action-key=true]').element.hasAttribute('disabled')).toBe(true)
        expect(dlg.element).toMatchSnapshot()
      })
    }, 5)
    let res = await dlg.wait()
    expect(res).toEqual({ msg: 'foo' })
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })
})
