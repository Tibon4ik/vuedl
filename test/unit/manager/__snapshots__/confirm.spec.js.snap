// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager Check confirm 1`] = `
<div
  class="dialog-layout"
>
  <div
    class="confirm-box"
  >
    <!---->
     
    <div
      class="confirm-message"
    />
     
    <div
      class="confirm-buttons"
    >
      <div
        class="column"
      >
        <button
          action-key="ok"
          class="button"
        >
          
        ok
      
        </button>
        <button
          action-key="cancel"
          class="button"
        >
          
        cancel
      
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`manager Check confirm with btns true|false 1`] = `
<div
  class="dialog-layout"
>
  <div
    class="confirm-box"
  >
    <!---->
     
    <div
      class="confirm-message"
    />
     
    <div
      class="confirm-buttons"
    >
      <div
        class="column"
      >
        <button
          action-key="true"
          class="button"
        >
          
        Yes
      
        </button>
        <button
          class="button"
        >
          
        No
      
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`manager returnable 1`] = `
<div
  class="dialog-layout"
>
  <p />
</div>
`;
